addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/handler/JeeLinkSensorHandler.java

```diff
@@ -0,0 +1,231 @@
+                    getThing().getUID(), channelUid.getId());
+        }
+    }
+
+    private void setNewSensorId(String oldId, String newId) {
+        ThingBuilder thingBuilder = editThing();
+        Map<String, String> map = new HashMap<>();
+        map.put(PROPERTY_SENSOR_ID, newId);
+        thingBuilder.withProperties(map);
+        updateThing(thingBuilder.build());
+
+        if (initialized) {
+            JeeLinkHandler jlh = (JeeLinkHandler) getBridge().getHandler();
+            JeeLinkReadingConverter<R> c = jlh.getConverter(clazz);
+            c.removeReadingListener(listener);
+            c.addReadingListener(newId, listener);
+        }
+        logger.info("Thing {} ({}) updated sensor id from {} to {}", getThing().getLabel(), getThing().getUID(), oldId,
+                newId);
+    }
+
+    @Override
+    public synchronized void initialize() {
+        if (initialized) {
+            logger.info("JeeLink sensor handler for thing {} ({}) is already initialized", getThing().getLabel(),
+                    getThing().getUID());
+            return;
+        }
+
+        statusUpdateJob = createStatusUpdateJob();
+        final int updateInterval = getConfig().as(JeeLinkSensorConfig.class).updateInterval;
+        if (updateInterval > 0) {
+            valueUpdateJob = createUpdateJob(updateInterval);
+        }
+
+        final int bufferSize = getConfig().as(JeeLinkSensorConfig.class).bufferSize;
+        String idStr = getConfig().as(JeeLinkSensorConfig.class).sensorId;
+        try {
+            updateStatus(ThingStatus.OFFLINE);
+
+            Average<R> avg = null;
+            if (bufferSize > 0) {
+                logger.debug("Using rolling average with buffer size {}...", bufferSize);
+                avg = new RollingReadingAverage<>(clazz, bufferSize);
+            }
+
+            logger.debug("Adding reading listener for id {}...", idStr);
+            listener = new JeeLinkReadingListener<R>(avg) {
+                @Override
+                public synchronized void handleReading(R reading) {
+                    try {
+                        if (isReadingWithinBounds(reading)) {
+                            boolean initial = lastReading == null;
+
+                            // propagate initial reading
+                            if (initial) {
+                                updateReadingStates(reading);
+                            }
+                            super.handleReading(reading);
+
+                            // propagate every reading in live mode
+                            if (!initial && updateInterval == 0) {
+                                updateReadingStates(getCurrentReading());
+                            }
+                        }
+
+                        // make sure status is online as soon as we get a reading
+                        updateStatus(ThingStatus.ONLINE);
+                    } catch (Throwable th) {
```

Comment:
Don't catch throwables: http://stackoverflow.com/questions/6083248/is-it-a-bad-practice-to-catch-throwable#answer-6083271